lib/matplotlib/pyplot.py

```diff
@@ -1228,11 +1249,29 @@ def subplot(*args, **kwargs):
         (ax for ax in fig.axes
          if hasattr(ax, 'get_subplotspec') and ax.get_subplotspec() == key),
         None)
+    # if we found an axes at the right position sort out if we
```

Comment:
the `ax = ax` below are a bit weird even though I get the logic.  Perhaps rewrite starting by removing the call to `next()` and then
```
for ax in fix.axes:
    if hasattr(ax, "...") and ax.get_subplotspec() == key:
        p_class, pkw = ...
        if <reuse>:
            break  # found it
else:  # didn't find an ax we can reuse, so make a new one
    ax = fig.add_subplot(...)
```

Reply:
Yup, I like that better.  I opted to do `ax=ax` rather than `pass` in the bodies of the if statements.